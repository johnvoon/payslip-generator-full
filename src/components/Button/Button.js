import React from "react";
import PropTypes from "prop-types";

const Button = ({
  children,
  customClassNames,
  type,
  handleClick,
  disabled
}) => {
  return (
    <button
      type={type}
      className={`btn text-uppercase ${customClassNames}`}
      onClick={handleClick}
      disabled={disabled ? disabled : false}
    >
      {children}
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.node.isRequired,
  customClassNames: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
  disabled: PropTypes.bool
};

export default Button;
