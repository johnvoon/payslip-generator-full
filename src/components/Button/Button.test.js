import React from "react";
import Button from "components/Button";
import { shallow } from "enzyme";

describe("Button", () => {
  let renderedComponent;
  const handleClick = jest.fn();

  beforeEach(() => {
    renderedComponent = shallow(
      <Button
        customClassNames="btn-primary"
        type="button"
        disabled={true}
        handleClick={handleClick}
      >
        Button
      </Button>
    );
  });

  it("should call handleClick when clicked", () => {
    renderedComponent.find("button").simulate("click");
    expect(handleClick).toHaveBeenCalled();
  });

  it("should match snapshot", () => {
    expect(renderedComponent).toMatchSnapshot();
  });

  it("should by default have disabled prop set to false if not supplied", () => {
    renderedComponent = shallow(
      <Button
        customClassNames="btn-primary"
        type="button"
        handleClick={handleClick}
      >
        Button
      </Button>
    );
    expect(renderedComponent.prop("disabled")).toBe(false);
  });
});
