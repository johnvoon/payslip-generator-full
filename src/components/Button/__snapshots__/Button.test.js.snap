exports[`Button should match snapshot 1`] = `
<button
  className="btn text-uppercase btn-primary"
  disabled={true}
  onClick={[Function]}
  type="button">
  Button
</button>
`;
